drone-fly-integration-tests/src/test/java/com/expediagroup/dataplatform/dronefly/core/integration/DroneFlyIntegrationTest.java

```diff
@@ -0,0 +1,194 @@
+     * complete.
+     * https://stackoverflow.com/questions/47312373/embeddedkafka-sending-messages-to-consumer-after-delay-in-subsequent-test
+     */
+    initEmbeddedKafka();
+    System.setProperty("instance.name", "test");
+    System.setProperty("apiary.bootstrap.servers", embeddedKafkaBroker.getBrokersAsString());
+    System.setProperty("apiary.kafka.topic.name", TOPIC);
+    System.setProperty("apiary.listener.list", "com.expediagroup.dataplatform.dronefly.core.integration.DummyListener");
+    initKafkaListener();
+
+    executorService.execute(() -> DroneFly.main(new String[] {}));
+    await().atMost(Duration.FIVE_MINUTES).until(DroneFly::isRunning);
+  }
+
+  @AfterEach
+  public void reset() {
+    DummyListener.reset();
+  }
+
+  @AfterAll
+  public void stop() throws InterruptedException {
+    DroneFly.stop();
+    executorService.awaitTermination(5, TimeUnit.SECONDS);
+  }
+
+  @Test
+  public void typical() {
+    AddPartitionEvent addPartitionEvent = new AddPartitionEvent(buildTable(), buildPartition(), true, hmsHandler);
+    kafkaMetaStoreEventListener.onAddPartition(addPartitionEvent);
+
+    CreateTableEvent createTableEvent = new CreateTableEvent(buildTable(), true, hmsHandler);
+    kafkaMetaStoreEventListener.onCreateTable(createTableEvent);
+
+    await().atMost(5, TimeUnit.SECONDS).until(() -> DummyListener.getNumEvents() > 1);
+
+    assertThat(DummyListener.getNumEvents()).isEqualTo(2);
+
+    ListenerEvent receivedEventOne = DummyListener.get(0);
+    ListenerEvent receivedEventTwo = DummyListener.get(1);
+
+    assertEvent(receivedEventOne, ADD_PARTITION);
+    assertEvent(receivedEventTwo, CREATE_TABLE);
+  }
+
+  private void assertEvent(ListenerEvent event, EventType eventType) {
+    assertThat(event.getStatus()).isTrue();
+
+    switch (eventType) {
+    case ADD_PARTITION:
+      assertThat(event).isInstanceOf(AddPartitionEvent.class);
+      AddPartitionEvent addPartitionEvent = (AddPartitionEvent) event;
+      assertThat(addPartitionEvent.getTable().getDbName()).isEqualTo(DATABASE);
+      assertThat(addPartitionEvent.getTable().getTableName()).isEqualTo(TABLE);
+      Iterator<Partition> iterator = addPartitionEvent.getPartitionIterator();
+      List<Partition> partitions = new ArrayList<>();
+      while (iterator.hasNext()) {
+        partitions.add(iterator.next());
+      }
+      assertThat(partitions).isEqualTo(Lists.newArrayList(buildPartition()));
+      assertThat(addPartitionEvent.getTable().getParameters()).isEqualTo(buildTableParameters());
+      break;
+    case CREATE_TABLE:
+      assertThat(event).isInstanceOf(CreateTableEvent.class);
+      CreateTableEvent createTableEvent = (CreateTableEvent) event;
+      assertThat(createTableEvent.getTable().getDbName()).isEqualTo(DATABASE);
+      assertThat(createTableEvent.getTable().getTableName()).isEqualTo(TABLE);
+      break;
+    default:
+      fail("Received an event with type other than ADD_PARTITION or CREATE_TABLE.");
```

Comment:
Add the eventType that was received to this failure  message so it's clear what went wrong without needing to get a debugger involved.

Reply:
done.